ReactQt/runtime/src/reactuimanager.cpp

```diff
@@ -164,31 +155,38 @@ void ReactUIManager::manageChildren(int containerReactTag,
                 child->setParentItem(container);
             }
 
-            // Add to layout
-            if (ReactAttachedProperties::get(child)->shouldLayout()) {
-                rfl->insertChild(i, child);
-                ReactFlexLayout::get(child)->setDirty(true);
-                ReactFlexLayout::get(child)->setParentItem(container);
+            auto containerFlexbox = Flexbox::findFlexbox(container);
+            auto childFlexbox = Flexbox::findFlexbox(child);
+            if (containerFlexbox && childFlexbox) {
+                containerFlexbox->addChild(i, childFlexbox);
             }
         }
-
-        rfl->setDirty(true);
     }
 
     m_bridge->visualParent()->polish();
 }
 
+//-----------------------------------------------------------------------------
+
 void ReactUIManager::replaceExistingNonRootView(int reactTag, int newReactTag) {
-    QQuickItem* item = m_views.value(reactTag);
-    if (item == nullptr) {
+    QQuickItem* oldItem = m_views.value(reactTag);
+    if (oldItem == nullptr) {
         qCritical() << __PRETTY_FUNCTION__ << "Attempting to access unknown item";
         return;
     }
 
-    QQuickItem* parent = ReactFlexLayout::get(item)->parentItem();
+    QQuickItem* parent = oldItem->parentItem();
     Q_ASSERT(parent != nullptr);
 
-    int itemIndex = ReactFlexLayout::get(parent)->getChildIndex(item);
+    int itemIndex = -1;
+    for (int i = 0; i < parent->childItems().count(); ++i) {
+        if (oldItem == parent->childItems().at(i)) {
```

Comment:
Looping across all items might affect performance ... with TicTacToe example I got loops with 2-3 iterations, but it seems the number of iterations can be any based on input JS. Maybe, makes sense to create ticket to address this later.

Reply:
Replaced loop with `indexOf()` function. that looks better, but don't think that it is faster.
I would suggest to create a ticket on this if we run into performance issues. So far, hardly can imagine UI where we have so many children that it impacts performance.

Reply:
@vkjr now it looks nicer at least! Thank you
We might discover such UI pretty soon :)